packages/client/src/App.re

```diff
@@ -1,12 +1,165 @@
+//         | SetLogin(isLoggedIn) => {...state, loggedIn: isLoggedIn}
+//         },
+//       initialState,
+//     );
+
+//   let handleDestinationSelect = destination =>
+//     dispatch(ChangeDestination(destination));
+
+//   let handlePositionSelect = station =>
+//     dispatch(SetCurrentPosition(station));
+
+//   let handleDroneInitResponse = data =>
+//     switch (data) {
+//     | Belt.Result.Ok(droneId) =>
+//       switch (droneId) {
+//       | Some(id) =>
+//         Js.log2("i app: ", id);
+//         dispatch(DroneId(id));
+//       | _ => ()
+//       }
+//     | Belt.Result.Error(e) => Js.log2("InitDroneError", e)
+//     };
+
+//   let handleDroneStatusSubscriptionData = data => {
+//     Js.log2("Drone subscription data:", data);
+//     ();
+//   };
+
+//   React.useEffect0(() => {
+//     let token = Auth.Storage.getLoginToken();
+//     switch (token) {
+//     | None => false->SetLogin |> dispatch
+//     | Some(_) => true->SetLogin |> dispatch
+//     };
+//     Some(_ => ());
+//   });
+
+//   let handleLogin = (payload: Auth.Payload.t) => {
+//     dispatch(SetLogin(true));
+//     Auth.Storage.setLoginToken(payload.token);
+//   };
+
+//   <div className="flex">
+//     {loggedIn ? React.null : <Login onLogin=handleLogin />}
+//     <div className="py-6 px-4 bg-blue-400 min-h-screen">
+//       <div className="w-full flex flex-col justify-center">
+//         <Icon name=`Dashboard className="text-gray-100 w-6 h-6 mb-6" />
+//       </div>
+//     </div>
+//     <div className="w-3/12 min-h-screen flex">
+//       <div className="w-full p-4 bg-white h-full flex flex-col">
+//         <label> {js|Från:|js}->React.string </label>
+//         <GeoSelectBox selectOptions=stations onChange=handlePositionSelect />
+//         <label> "Till:"->React.string </label>
+//         <Destination handleDestinationSelect />
+//         {switch (currentPosition, currentDestination) {
+//          | (Some(start), Some(stop)) =>
+//            <InitDrone start stop handleDroneInitResponse />
+//          | _ => React.null
+//          }}
+//         {switch (droneId) {
+//          | Some(id) => <DronePosition id handleDroneStatusSubscriptionData />
+//          | _ => React.null
+//          }}
+//       </div>
+//     </div>
+//     <div className="w-9/12 bg-gray-400 h-12 relative min-h-screen">
+//       <Map ?currentPosition ?currentDestination> <div /> </Map>
+//     </div>
+//   </div>;
+// };
 [@react.component]
 let make = () => {
   let url = ReasonReactRouter.useUrl();
 
-  switch (url.path) {
-  | [] => <Start />
-  | ["boka-resa"] => <Book />
-  | ["resor"] => <Trips />
-  | ["resa", id] => <Trip id />
-  | _ => <p> "404. Not found"->React.string </p>
+  let (loggedIn, setLoggedIn) = React.useState(_ => false);
+
+  React.useEffect0(() => {
+    let token = Auth.Storage.getLoginToken();
+    switch (token) {
+    | None => setLoggedIn(_ => false)
+    | Some(_) => setLoggedIn(_ => true)
+    };
+    Some(_ => ());
```

Comment:
Could do:

```re
let hasToken = Auth.Storage.getLoginToken()->Belt.Option.isSome;
setLoggedIn(_prevValue => hasToken);

None;
```

Since we don't need to unsubscribe to something we can return `None` to `useEffect0`.